### PR TITLE
[plugin.video.arloview] 1.0.0

### PR DESCRIPTION
**Pull request by JavaWiz1**
### Description

Stream your Arlo security cameras to kodi interface
### Checklist:

- [x ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [ x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,5 @@
 **/*.travis.yml
 **/*.xcf
 **/*.zip
+
+**/*.vscode
```
Comment (Rechi) [inline .gitignore]: remove this change

File: plugin.video.arloview/resources/lib/arlo_stream.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            return
+        xbmc.log("[{}-{}] {}".format(ADDON_ID, ADDON_VERSION, msg), level)
+
+    def check_first_run(self):
+
+        if REAL_SETTINGS.getSetting("userid") == "":
+            self.log("check_first_run()- userid NOT set!", xbmc.LOGNOTICE)
+            msg = "Set ARLO credentials, otherwise...\nVideo will not stream correctly!"
+            icon = REAL_SETTINGS.getAddonInfo('icon')
+            xbmc.executebuiltin('Notification(%s, %s, %d, %s)' % (ADDON_NAME, msg, 5000, icon))
```
Comment (Rechi) [inline plugin.video.arloview/resources/lib/arlo_stream.py]:
```suggestion
            xbmcgui.Dialog().notification(ADDON_NAME, msg, icon, 5000)
```

File: plugin.video.arloview/resources/lib/arlo_stream.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+        while is_playing:
+            # self.log("Player is playing...", xbmc.LOGDEBUG)
+            time.sleep(1)
+            is_playing = xbmc.Player().isPlaying()
+            if not is_playing:
+                self.log("Player has stopped.", xbmc.LOGDEBUG)
+
+        # Update camera details after viewing (battery strength, signal strength,...)
+        self._update_arlo_cameras_details()
```
Comment (Rechi) [inline plugin.video.arloview/resources/lib/arlo_stream.py]: I don't see a place where the update details are use, as this is the last command executed and it doesn't persist the info.

File: .gitignore [hunk 4]
```diff
@@ -17,3 +17,4 @@
 **/*.travis.yml
 **/*.xcf
 **/*.zip
+
```
Comment (Rechi) [inline .gitignore]: also remove the empty line addition

**Comment by Rechi** [comment]
> Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Please squash your commits and it's ready to go.

**Comment by JavaWiz1** [comment]
Thank you for your patience and help on this. Change made as requested, I think we are good to go.. Thanks again.

________________________________

File: plugin.video.arloview/resources/lib/arlo_stream.py [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        camera = self._get_camera(device_id)
+        stream_url = self.arlo.StartStream(self.basestation, camera)
+
+        # Create a playable item with a path to play.
+        play_item = xbmcgui.ListItem(path=stream_url, label=camera['deviceName'])
+        play_item.setProperty("isPlayable", "true")
+        # Pass the item to the Kodi player.
+        xbmcplugin.setResolvedUrl(self._handle, True, listitem=play_item)
+
+        wait_secs = 5
```
Comment (Rechi) [inline plugin.video.arloview/resources/lib/arlo_stream.py]: As you have now removed the `self._update_arlo_cameras_details()` call below, this block (checking play state) doesn't do anything useful.

Comment (Rechi) [reply plugin.video.arloview/resources/lib/arlo_stream.py]: Waiting for player to start playing also is of no use. You only log that player has started, but Kodi already logs that.

**Pull request merged**